### PR TITLE
Use MySQL for IT tests to better approximate prod

### PR DESCRIPTION
**Pull request by callunity**

### CODE REVIEW

File: miso-web/pom.xml [hunk 1]
```diff
@@ -503,35 +569,6 @@
 	        </executions>
 	      </plugin>
 	      <plugin>
-	        <!-- create H2 in-memory DB, accessible via TCP -->
-	        <groupId>net.xeric.maven</groupId>
-	        <artifactId>h2-maven-plugin</artifactId>
```
Comment (djcooke) [inline miso-web/pom.xml]: Can remove this from pluginManagement in the parent pom as well

File: miso-web/src/it/resources/db/migration/V9000__integration-test_data.test.sql [hunk 2]
```diff
@@ -1,6 +1,8 @@
 -- fixes LibraryDesign changes that were marked "NoTest" in V0130
-ALTER TABLE LibraryDesign DROP CONSTRAINT `uk_libraryDesign_name`;
-ALTER TABLE LibraryDesign ADD CONSTRAINT `uk_libraryDesign_name_sampleClass` UNIQUE (`name`, `sampleClassId`);
+--ALTER TABLE LibraryDesign DROP FOREIGN KEY `uk_libraryDesign_name`;
+--ALTER TABLE LibraryDesign ADD CONSTRAINT `uk_libraryDesign_name_sampleClass` UNIQUE (`name`, `sampleClassId`);
```
Comment (djcooke) [inline miso-web/src/it/resources/db/migration/V9000__integration-test_data.test.sql]: remove commented-out lines

File: miso-web/src/it/resources/it-context.xml [hunk 3]
```diff
@@ -30,6 +30,11 @@
 
   <bean id="sessionFactory" class="org.springframework.orm.hibernate4.LocalSessionFactoryBean">
     <property name="dataSource" ref="dataSource" />
+    <property name="hibernateProperties">
+      <props>
+        <prop key="hibernate.dialect">org.hibernate.dialect.MySQL5InnoDBDialect</prop>
```
Comment (djcooke) [inline miso-web/src/it/resources/it-context.xml]: Shouldn't need to specify this, but in the version of Hibernate we're using (4.2), we should probably use MySQL57InnoDBDialect if we do specify

File: pom.xml [hunk 4]
```diff
@@ -465,6 +465,11 @@
         <version>${jackson.version}</version>
       </dependency>
       <dependency>
+        <groupId>io.fabric8</groupId>
+        <artifactId>docker-maven-plugin</artifactId>
+        <version>0.21.0</version>
```
Comment (djcooke) [inline pom.xml]: Did you try the latest version (0.22.1)?

**Comment by callunity** [comment]
And I :heart: it!

**Pull request merged**